File: PassTask/3.1P/Clock/Counter.cs
namespace ClockProgram
{
    public class Counter
    {
        private int _count;
        private string _name;
        public Counter(string name)
        {
            _name = name;
            _count = 0;
        }

        public void Increment()
        {
            _count += 1;
        }

        public void Reset()
        {
            _count = 0;
        }
        public void ResetByDefault()
        {
            // Answer for 13
            // _count = 2147483647041;
            // above value create the crash(error) for the compile time
            // due to overflow of integer which is _count
            // to run safetly (However, _count will get unexcpeted value), we can use
            // uncheckd block, which neglecte the overflow exception throw and
            // use explictly type cast to in
            // becuase 2147483647041 is bigger than int max value
            unchecked
            {
                _count = (int)2147483647041;
            }
        }

        public string Name
        {
            get { return _name; }
            set { _name = value; }
        }

        public int Ticks
        {
            get { return _count; }
        }
    }
}

File: PassTask/11.1P/Clock/counter.py
class Counter:
    def __init__(self, name):
        self._name = name
        self._count = 0

    def increment(self):
        self._count += 1

    def reset(self):
        self._count = 0

    def reset_by_default(self):
        # Original large value
        large_value = 2147483647041

        # Simulate int32 overflow behavior
        int32_max = 2147483647
        int32_min = -2147483648
        int32_range = int32_max - int32_min + 1

        # This simulates the unchecked overflow behavior from C#
        self._count = ((large_value - int32_min) % int32_range) + int32_min

    @property
    def name(self):
        return self._name

    @name.setter
    def name(self, value):
        self._name = value

    @property
    def ticks(self):
        return self._count
